src/jshint.js

```diff
@@ -2002,6 +2035,19 @@ var JSHINT = (function() {
     error("E014");
   };
   assignop("%=", "assignmod", 20);
+  assignop("**=", function(left, that) {
+    if (!state.inES7()) {
```

Comment:
Seems like you might consider renaming `inES7()` to match `esversion` values, I don't think maintaining api compat is a big deal for jshint. It's awkward to have two conventions in the same project.